core/server/api/v2/preview.js

```diff
@@ -0,0 +1,41 @@
+const common = require('../../lib/common');
+const models = require('../../models');
+const ALLOWED_INCLUDES = ['author', 'authors', 'tags'];
+
+module.exports = {
+    docName: 'preview',
+
+    read: {
+        permissions: true,
+        options: [
+            'include'
+        ],
+        data: [
+            'uuid'
+        ],
+        validation: {
+            options: {
+                include: {
+                    values: ALLOWED_INCLUDES
+                }
+            },
+            data: {
+                uuid: {
+                    required: true
+                }
+            }
+        },
+        query(frame) {
+            return models.Post.findOne(Object.assign({status: 'all'}, frame.data), frame.options)
+                .then((model) => {
+                    if (!model) {
+                        throw new common.errors.NotFoundError({
```

Comment:
Just for clarity when we migrate more controllers. By design should we be throwing `throw new common.errors.NotFoundError` in the query level or in output serializer level? The latter seems like a better place because that's how 'empty' data would be formatted but definitely see the use case in controllers where you have more logic than just a simple query and return

Reply:
Haven't thought about it yet. IMO the model layer should throw this error by default. But nothing we have fully decided yet, because we have a mixture of usages.